File: templates/node/example/handler.mjs
#!/usr/bin/env node
// SPDX-License-Identifier: Apache-2.0
// Example cfunc handler in Node. Resolves the cfunc SDK via NODE_PATH —
// register with: env=["NODE_PATH=/path/to/sdks/node"]
import { start, Response } from 'cfunc'

await start(async (event, ctx) => {
  return new Response({
    status: 200,
    headers: { 'Content-Type': 'application/json' },
    body: {
      hello: 'world',
      method: event.method,
      path: event.path,
      lang: 'node',
      node_version: process.version,
    },
  })
})

File: templates/python/example/handler.py
#!/usr/bin/env python3
"""Example cfunc handler in Python.

The SDK module (cfunc.py) is expected on PYTHONPATH. The gateway/runtime
arranges this by mounting the SDK layer at /opt/cfunc-sdk; for local
runs, set PYTHONPATH=$REPO/sdks/python.
"""
import cfunc


def handle(event: cfunc.Event, ctx: cfunc.Context) -> cfunc.Response:
    return cfunc.Response(
        status=200,
        headers={"Content-Type": "application/json"},
        body={
            "hello": "world",
            "method": event.method,
            "path": event.path,
            "lang": "python",
        },
    )


if __name__ == "__main__":
    cfunc.start(handle)
